### ISSUE TITLE
archive preview via `vim/nvim`

### ISSUE THREAD

**Issue by asklow** [NikitaIvanovV/ctpv#22]
When directly opened via vim / nvim, they show file/directory list. But in `lf`, archives are considered `binary` bydefault. Currently, `atool` is the only archive previewer for ctpv (which looks bad with permissions & date columns first).
Can these editors be used as archive previews ? Caz they obviously able to open them & list items.

**Comment by NikitaIvanovV** [comment]
> Can these editors be used as archive previews ? Caz they obviously able to open them & list items.

You can add this to `~/.config/ctpv/config`

```
preview vimarchive .zip .tar.gz {{
    # Save file
    vim -c "w! $cache_f | q" "$f" >/dev/null 2>&1

    # Remove first comments from file
    sed '1,4d' "$cache_f"
}}
```

Becuase vim is a big program, it's kinda slow, so I will probably not add this as a built-in preview.

**Comment by NikitaIvanovV** [comment]
Can't reproduce, even when using nvim. It's a very dirty solution to use vim to create a preview anyway, there must be some simpler program we could use for it. What about writing all the previews for all the archive types manually without using atool? Sounds like a lot of work...

**Comment by NikitaIvanovV** [comment]
I've tried it and it didn't quite work. The problem is that atool is just a wrapper around different archive programs, so order of columns and general format of the output depends on which archive type you are trying to preview.

**Comment by NikitaIvanovV** [comment]
> Even though using vim/nvim is a dirty way, it's the only sane way besides atool right.

I refuse to believe it's the only sane way, there must something better (e.g., my suggestion to write a preview manually for each type (or only the most common ones)).

> it's as fast as atool's preview despite the editor itself being large.

I have a very cheap laptop and a preview using vim is noticably slow on it.

> almost all terminal users have either one of them handy.

That's true.

**Comment by NikitaIvanovV** [comment]
I agree that atool doesn't produce the best result, so I will keep the issue open while a better solution is found.
